sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -1966,8 +1966,12 @@ private Task TrackPublishHandlerAsActiveAsync(Task handlerTask)
 
             _activePublishingHandlers.TryAdd(handlerTask, 0);
 
-            var continationTask = handlerTask.ContinueWith((runTask, trackedTask) => _activePublishingHandlers.TryRemove((Task)trackedTask, out _), handlerTask, TaskScheduler.Default);
-            return continationTask;
+            var continuationTask = handlerTask.ContinueWith(static (runTask, state) =>
+            {
+                var (trackedTask, activeHandlers) = (Tuple<Task, ConcurrentDictionary<Task, byte>>)state;
+                return activeHandlers.TryRemove(trackedTask, out _);
+            }, Tuple.Create(handlerTask, _activePublishingHandlers), CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
+            return continuationTask;
```

Comment:
```suggestion
            var continuationTask = handlerTask.ContinueWith(static (runTask, state) =>
            {
                var (trackedTask, activeHandlers) = (Tuple<Task, ConcurrentDictionary<Task, byte>>)state;
                return activeHandlers.TryRemove(trackedTask, out _);
            }, 
            Tuple.Create(handlerTask, _activePublishingHandlers), CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
            
            return continuationTask;
```

nit: Adding some whitespace for readability.